### ISSUE TITLE
Spatial Extents

### ISSUE THREAD

**Issue by nalarsson** [julieshortridge/Climate-Analysis#3]

**Comment by nalarsson** [comment]
Spatial Data with Attributes for Desired Extent Acquired
[States.pdf](https://github.com/julieshortridge/Climate-Analysis/files/10873778/States.pdf)

**Comment by nalarsson** [comment]
![States](https://user-images.githubusercontent.com/117185142/222497053-71bd7674-c840-4608-9511-1fbfbadf5e4b.png)

**Comment by nalarsson** [comment]
Creating summary tables for data. These are for the UKESM1-0-LL historical data for year 1950. There is no tasmax data for this model; let me know if you want me to make a tasmax summary for a different model/emissions scenario or if you want to download a tasmax file from this model and I can do a summary on that.

[summary_tasmin.csv](https://github.com/julieshortridge/Climate-Analysis/files/10914374/summary_tasmin.csv)
[summary_pr.csv](https://github.com/julieshortridge/Climate-Analysis/files/10914376/summary_pr.csv)

**Comment by nalarsson** [comment]
Error with ncvar_get() command.

>  ncvar_get(ncin, varid="pr")
Error in R_nc4_inq_varndims: NetCDF: Not a valid ID
Error in ncvar_ndims(ncid, varid) : error returned from C call

ncin is the variable that stores the nc_open command. No issues with this command.

This was the error I tried to show on Friday, but then I ran my code again and it wasn't producing the error. I've tried running the code on a few files to see if a single file was corrupted, but that isn't the case. I've also confirmed that the "pr" variable does exist in the file I'm working with; I've run names(ncin$var) and "pr" is returned, and I have visually confirmed that data exists in this file.

This is preventing me from creating summaries of files with units we can understand, but I am able to keep working on other aspects of the NetCDF tutorial.

**Comment by nalarsson** [comment]
Just so you're aware, time is "days since 1850-01-01" in CMIP6. In CMIP5, it was "days since 1860-01-01". Doesn't seem like it will be an issue but I took note just in case.

**Comment by nalarsson** [comment]
Issues with selecting correct cell from raster. Here's what I have checked so far. I'm stumped, let me know if you have any advice. 

Checked extents of raster and county shapefile:
> extent(tmin.raster)
class      : Extent 
xmin       : -90 
xmax       : -73 
ymin       : 30 
ymax       : 43 
> extent(app.counties)
class      : Extent 
xmin       : -90.3103 
xmax       : -75.24227 
ymin       : 32.0346 
ymax       : 41.97752 


Confirmed accuracy in logic for selecting raster
bb.lon
[1] -80.41394
> bb.xcoord <- max(which(lon < bb.lon.east))
> bb.xcoord
[1] 38
> lon[bb.xcoord]
[1] 279.375
> lon[bb.xcoord]-360
[1] -80.625
> bb.lat
[1] 37.22957
> bb.ycoord <- max(which(lat < bb.lat))
> bb.ycoord
[1] 29
> lat[bb.ycoord]
[1] 37.125

Lat/Lon for Montgomery County, which is highlighted in blue on the map. 

Montgomery.County@bbox         
       min       max 
x -80.61406 -80.17640 
y  36.98157  37.36679


First photo: What I think is the more correct one.
![CheckSpatialExtent](https://user-images.githubusercontent.com/117185142/224110326-a724f69b-1a2c-463c-8f57-3e4454245dc1.png)

Second photo: Switched lat/lon to make sure I didn't mess that up.
![CheckSpatialExtentError](https://user-images.githubusercontent.com/117185142/224112306-c319453e-f117-4cf8-8b62-ad1132917e14.png)
